python/ray/data/read_api.py

```diff
@@ -1521,7 +1547,8 @@ def from_pandas(
 
 @DeveloperAPI
 def from_pandas_refs(
-    dfs: Union[ObjectRef["pandas.DataFrame"], List[ObjectRef["pandas.DataFrame"]]]
+    dfs: Union[ObjectRef["pandas.DataFrame"], List[ObjectRef["pandas.DataFrame"]]],
+    logical_op: Optional[FromPandasRefs] = None,
```

Comment:
sorry on a second thought, I think we shouldn't expose `logical_op` to users in any `DeveloperAPI`/`PublicAPI`. can we work around something like:

```py
@PublicAPI
def from_modin(df: "modin.DataFrame") -> MaterializedDatastream[ArrowRow]:
  ds = from_pandas_refs(parts)
  ds._logical_plan = FromModin(df)
  return ds

@DeveloperAPI
def from_pandas_refs(
    dfs: Union[ObjectRef["pandas.DataFrame"], List[ObjectRef["pandas.DataFrame"]]]
) -> MaterializedDatastream[ArrowRow]:
  ...
```

Reply:
Yeah, a workaround like this works to update the `ds`'s underlying logical plan:
```
def from_modin(df: "modin.DataFrame") -> MaterializedDatastream[ArrowRow]:
    parts = unwrap_partitions(df, axis=0)
    ds = from_pandas_refs(parts)
    ds._plan.link_logical_plan(LogicalPlan(FromModin(df)))
    return ds
```